micropipenv.py

```diff
@@ -150,6 +151,20 @@ class NotSupportedError(MicropipenvException):
     """Raised when the given feature is not supported by micropipenv."""
 
 
+def _import_toml():  # type: () -> Any
+    """Imports and returns toml or pytoml (in this order)"""
+    for module in "toml", "pytoml":
+        try:
+            return import_module(module)
+        except ImportError:
+            pass
+    else:
+        raise ExtrasMissing(
+            "Failed to import toml needed for parsing Pipfile, please install micropipenv "
+            "with toml extra: pip install micropipenv[toml]"
+        )
```

Comment:
BTW I would suggest to unroll the loop and use exception chaining instead of silently passing the import error:

```python
    try:
        return import_module("toml")
    except ImportError:
        try:
            return import_module("pytoml")
        except ImportError as exc:
            raise ExtrasMissing(
                "Failed to import toml needed for parsing Pipfile, please install micropipenv "
                "with toml extra: pip install micropipenv[toml]"
            ) from exc
```

Reply:
What is the benefit of the exception chaining in this case? I honestly like my approach more because it's flatter.

Reply:
It states all the steps that were done to look for toml support - possibly easier for debugging. If you want to keep it the way as is, it's ok.

https://www.python.org/dev/peps/pep-3134/#motivation

Reply:
Yes, you are right. Exceptions chaining makes perfect sense. The problem, in this case, is that raising ExtrasMissing from the previous Importerror might mean that the traceback would contain something like `ModuleNotFoundError: No module named 'pytoml'` which would send a different message ("we need pytoml") than the ExtrasMissing error message ("we need toml").

There is a solution for this, of course, but I think that the flatter code is better here, the exception occurs only on the one place in code and it's straightforward to debug.